spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -394,15 +397,98 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     }
   }
 
-  test("test coalesce lazy eval") {
-    withSQLConf(
-      SQLConf.ANSI_ENABLED.key -> "true",
-      CometConf.COMET_EXPR_ALLOW_INCOMPATIBLE.key -> "true") {
-      val data = Seq((9999999999999L, 0))
-      withParquetTable(data, "t1") {
+  test("ANSI support for add") {
```

Comment:
These tests use integer inputs, but the implementation supports both integer and float inputs, so we should have tests for both cases.

Reply:
Thank you for the feedback . As discussed on a different thread, Spark's ANSI arithmetic checks for add , subtract, multiply and divide only support Integer overflows. I updated code to handle division use case (where operands are always cast as floats and an error is thrown only if the dividend is 0 )